Sources/Workspace/ManagedArtifact.swift

```diff
@@ -70,6 +80,21 @@ extension Workspace {
                 path: path
             )
         }
+        
+        /// Create an artifact extracted from a local archive path.
+        public static func localArchived(
+            packageRef: PackageReference,
+            targetName: String,
+            path: AbsolutePath,
+            archivePath: AbsolutePath
+        ) -> ManagedArtifact {
+            return ManagedArtifact(
+                packageRef: packageRef,
+                targetName: targetName,
+                source: .localArchived(archivePath: archivePath),
+                path: path
+            )
+        }
```

Comment:
could removing the special type for archived local artifact simplify the solution?

For example, if we do not add `localArchived` and instead allow `local` to take in a directory or a zip file, then in the workspace code where we handle the artifacts we would check for if the artifact path is a directory or a zip file we could probably achieve the same with less impact?

Reply:
Thanks a lot for the feedback, now when I know the additional constraints (minimizing the API impact), I'll try to optimize the solution for that.

Reply:
Done, there are zero API changes now, and only the private implementations are affected.